From 8b9d88f647ece27985f8327d217a576e089bfc25 Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Fri, 8 Jun 2018 10:43:36 +0000
Subject: [PATCH] TRITON-478 agent builds fail as non-root when postinstall.sh
 writes to / tweak message, move checks to after trace setup

---
 agent/pkg/postinstall.sh | 6 +++++-
 relay/pkg/postinstall.sh | 7 ++++++-
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/agent/pkg/postinstall.sh b/agent/pkg/postinstall.sh
index a3b8588..a6b7f23 100755
--- a/agent/pkg/postinstall.sh
+++ b/agent/pkg/postinstall.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -28,6 +28,10 @@ fi
 set -o errexit
 set -o pipefail
 
+if [[ -z "$npm_config_smfdir" || ! -d "$npm_config_smfdir" ]]; then
+    echo "Skipping amon-agent postinstall (assuming build-time install)"
+    exit 0
+fi
 
 TOP=$(cd $(dirname $0)/../ >/dev/null; pwd)
 
diff --git a/relay/pkg/postinstall.sh b/relay/pkg/postinstall.sh
index d0336af..b375032 100755
--- a/relay/pkg/postinstall.sh
+++ b/relay/pkg/postinstall.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 OS=$(uname -s)
@@ -18,6 +18,11 @@ fi
 
 set -o xtrace
 
+if [[ -z "$npm_config_smfdir" || ! -d "$npm_config_smfdir" ]]; then
+  echo "Skipping amon-relay postinstall (assuming build-time install)"
+  exit 0
+fi
+
 export SMFDIR=$npm_config_smfdir
 
 function subfile () {
-- 
2.21.0

